modules/core/src/main/scala/org/scalasteward/core/coursier/CoursierAlg.scala

```diff
@@ -90,20 +87,12 @@ object CoursierAlg {
             result <- maybeFetchResult.toOptionT[F]
             moduleVersion = (coursierDependency.module, coursierDependency.version)
             (_, project) <- result.resolution.projectCache.get(moduleVersion).toOptionT[F]
-            url <- getScmUrlOrHomePage(project.info)
-              .toOptionT[F]
-              .orElse(OptionT(getParentArtifactUrl(project)))
+            parentUrl = getParentDependency(project)
```

Comment:
Will this trigger the fetching of the parent project metadata also in case where it is not needed ?

Reply:
No, since `parentUrl` is not `flatMap`ped via `<-` in the for-comprehension. And since the code here is referentially transparent `x.orElse(y)` is the same as `val a = y; x.orElse(a)`.

Reply:
One marginal "downside" is that `getParentDependency(project).traverseFilter(getArtifactUrlImpl(_, resolvers))` is now always evaluated (but not executed) even if the parent POM is not needed. I think I'll change that again to make it clearer. Thanks for the nudge, @mzuehlke!